Infodev doc review (#944)
IntelPython/dpctl: docs/docfiles/user_guides/manual/dpctl/basic_concepts.rst

.. note:: For SYCL-level details, refer to a more topical SYCL* reference, such as the :sycl_spec_2020:`SYCL 2020 spec <>`.

* **Heterogeneous computing**
    Refers to using multiple devices in a program.

* **Host**
    Every program starts by running on a host, and most of the lines of code in
    a program, in particular lines of code implementing the Python* interpreter
    itself, are usually for the host. Hosts are customarily CPUs.

* **Device**
    A device is an XPU connected to a host that is programmable with a specific
    device driver. Different types of devices can have different architectures
    (CPUs, GPUs, FPGA, ASICs, DSP) but are programmable using the same
    :oneapi:`oneAPI <>` programming model.

* **Platform**
    A device driver installed on the system is called the platform. As multiple
    devices of the same type can share the same device driver, a platform may
    contain multiple devices. The same physical hardware (for example, GPU)
    may be reflected as two separate devices if they can be programmed by more
    than one platform. For example, the same GPU hardware can be listed as an

>     OpenCL* GPU device and a Level-Zero* GPU device.

Reviewer: ```suggestion
    OpenCL\* GPU device and a Level-Zero\* GPU device.
```

* **Context**
   Holds the runtime information needed to operate on a device or a
   group of devices from the same platform. Contexts are relatively expensive
   to create and should be reused as much as possible.

* **Queue**
   A queue is needed to schedule the execution of any computation or data
   copying on the device. Queue construction requires specifying a device

>    and contextual targeting that device as well as additional properties,

Reviewer: ```suggestion
   and a context targeting that device as well as additional properties,
```

Here context is the term, like in the context described in the previous item.

   such as whether profiling information should be collected or submitted
   tasks are executed in the order in which they were submitted.

* **Event**
   Holds information related to computation or data movement operation
   scheduled for the execution on a queue. That can be its execution status or

>    profiling information if the queue, to which the task is submitted, is allowed

Reviewer 2: ```suggestion
   profiling information of the queue to which the task is submitted.
```

>    for the collection of such information. Events can be used to specify task

Reviewer 2: ```suggestion
Events can be used to specify task
```
